src/Rabbitmq/Manager/Manager.php

```diff
@@ -109,32 +114,43 @@ public function addCommand(\Maleficarum\Command\AbstractCommand $command, string
 
         // check if the specified connection identifier exists
         if (!array_key_exists($connectionIdentifier, $this->connections)) throw new \InvalidArgumentException(sprintf('Provided connection identifier does not exist. %s', __METHOD__));
-        
+
         // recover the specified connection for internal storage
         $connection = $this->connections[$connectionIdentifier]['connection'];
 
         // initialise the connection if necessary
         $connection->connect();
 
+        // get publish confirm timeout for connection
+        $publishConfirmTimeout = $this->connections[$connectionIdentifier]['publishConfirmTimeout'];
+
         $applicationHeaders = \Maleficarum\Ioc\Container::get('PhpAmqpLib\Wire\AMQPTable', [$commandHeaders]);
 
         // send the command to the message broker
         $message = \Maleficarum\Ioc\Container::get(
             'PhpAmqpLib\Message\AMQPMessage',
             [$command->toJSON(), ['delivery_mode' => 2, 'application_headers' => $applicationHeaders]]);
         $channel = $connection->getChannel();
+
+        // establish publish confirm mode if necessary
+        $publishConfirmTimeout > 0 and $channel->confirm_select();
+
         $channel->basic_publish($message, $connection->getExchangeName(), $connection->getQueueName());
+
+        // wait for publish confirm
+        $channel->wait_for_pending_acks($publishConfirmTimeout);
```

Comment:
If the confirm_select() is conditional why isn't wait_for_pending_acks() conditional as well?